project/src/main/webapp/common.js

```diff
@@ -0,0 +1,38 @@
+//Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// use modern JavaScript (ES5)
+"use strict"
+
+function addCommonElements() {
+  addNavbar();
+  addHeader();
+}
+
+function addNavbar() {
+  let navbar = `<a href="/index.html">Home</a>
+      <br>
+      <a href="/">Resources</a>
+      <br>
+      <a href="/">Polling Locations</a>
+      <br>`;
```

Comment:
The code may be more maintainable if you use JavaScript to create elements, instead of putting everything in a String. Especially if the navbar HTML structure will change/expand in the future.
If the HTML structure will unlikely change and the hrefs are the only thing that will be filled out, then using a String is more convenient.

Reply:
Using the JS to create elements this long tends to make a bit of an unreadable mess. I think my plan is to create elements for the dynamic parts, but leave the hardcoded links as a string.

Reply:
Ok! If this section will remain the fixed structure, I agree it's easier to have a hard-coded string. You can fill in the links later with string formatting.